src/app/clusters/basic/basic.cpp

```diff
@@ -83,26 +83,85 @@ class PlatformMgrDelegate : public DeviceLayer::PlatformManagerDelegate
 
 PlatformMgrDelegate gPlatformMgrDelegate;
 
-} // anonymous namespace
-
-void emberAfBasicClusterServerInitCallback(chip::EndpointId endpoint)
+class BasicAttrAccess : public app::AttributeAccessInterface
 {
-    EmberAfStatus status;
-
-    char nodeLabel[DeviceLayer::ConfigurationManager::kMaxNodeLabelLength + 1];
-    if (ConfigurationMgr().GetNodeLabel(nodeLabel, sizeof(nodeLabel)) == CHIP_NO_ERROR)
+public:
+    BasicAttrAccess() : app::AttributeAccessInterface(Optional<EndpointId>::Missing(), app::Clusters::Basic::Id) {}
+    CHIP_ERROR Read(const app::ConcreteReadAttributePath & aPath, app::AttributeValueEncoder & aEncoder) override
     {
-        status = Attributes::NodeLabel::Set(endpoint, chip::CharSpan(nodeLabel, strlen(nodeLabel)));
-        VerifyOrdo(EMBER_ZCL_STATUS_SUCCESS == status, ChipLogError(Zcl, "Error setting Node Label: 0x%02x", status));
+        if (aPath.mAttributeId == app::Clusters::Basic::Attributes::NodeLabel::Id)
+        {
+            char nodeLabel[DeviceLayer::ConfigurationManager::kMaxNodeLabelLength + 1];
+            if (ConfigurationMgr().GetNodeLabel(nodeLabel, sizeof(nodeLabel)) == CHIP_NO_ERROR)
```

Comment:
Should an error be return otherwise or it's handled by another layer?

Reply:
I have tested this case, if I return an error here, the server in fact will return nothing and the controller will be timeout. While if I return success without any Encode, the controller will get null data soon. 
For me, I think do not return is better because we need not wait and can get errors. Also, many other examples are not returning errors.

Of course, it will be much better to return an error to the controller if here returns an error. It is on my want list, but I do not know who should talk to about this.

Reply:
There are TODO comments in ember-compatibility-functions about handling error returns from this code better...

Reply:
OK, I will change it.